File: C#/Toolkit.CreatePDF.cs

using System;
using System.Text;

namespace ToolkitExamples
{
    class Program
    {
        static void Main(string[] args)
        {
            string strPath = System.AppDomain.CurrentDomain.BaseDirectory;

            // Starting with Toolkit version 10 native DLLs are no longer
            // copied to the system folder. The Toolkit constructor must
            // be called with the path to the native DLLs or place them
            // in your applications working directory. This example
            // assumes they are located in the default installation folder.
            // (Use x86 in the path for 32b applications)
            string toolkitPath = $@"{Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles)}\ActivePDF\Toolkit\bin\x64";

            // Instantiate Object
            using (APToolkitNET.Toolkit toolkit = new APToolkitNET.Toolkit(CoreLibPath: toolkitPath))
            {
                // Set the PDF page Height and Width (72 = 1")
                toolkit.OutputPageHeight = 792.0f;
                toolkit.OutputPageWidth = 612.0f;

                // Create the new PDF file
                int result = toolkit.OpenOutputFile(FileName: $"{strPath}Toolkit.NewPDF.pdf");
                if (result != 0)
                {
                    WriteResult($"Error opening output file: {result.ToString()}", toolkit);
                    return;
                }

                // Each time a new page is required call NewPage
                toolkit.NewPage();

                // Get the current version of Toolkit and save it to print on
                // the PDF
                string toolkitVersion = toolkit.ToolkitVersion;

                // Text can be added onto the new page with
                // SetFont, PrintText and PrintMultilineText functions
				float font_size = 24.0f;
                string font_name = "Times New Roman";
                toolkit.SetFont(FontName: font_name, FontSize: font_size);
				
				float y = 720.0f;
                toolkit.PrintText(X: 72.0f, Y: y, 1, $"Toolkit Version: {toolkitVersion}", font_name, font_size);
                toolkit.PrintText(X: 72.0f, Y: y - 25, 1, $"by ActivePDF - An Apryse Company", font_name, font_size);

                // Images can be added onto the new page with PrintImage,
                // PrintJPEG and PrintTIFF
                // Substitute your images here
		toolkit.PrintImage(
			ImageFileName: strPath + "butterfly.png",
			X: 72.0f,
			Y: 500.0f,
			Width: 100.0f,
			Height: 100.0f,
			PersistRatio: true,
			PageNumber: 0);

		toolkit.PrintImage(
			ImageFileName: strPath + "logo_red.png",
			X: 200.0f,
			Y: 500.0f,
			Width:100.0f,
			Height: 100.0f,
			PersistRatio: true,
			PageNumber: 0);

		toolkit.PrintJPEG(
			FileName: strPath + "dice.jpg",
			X: 72.0f,
			Y: 400.0f,
			Width: 100.0f,
			Height: 100.0f,
			PersistRatio: true,
			PageNumber: 0);

		toolkit.PrintTIFF(
			FileName: strPath + "Toolkit.Input.tif",
			X: 200.0f,
			Y: 400.0f,
			Width: 100.0f,
			Height: 100.0f,
			PersistRatio: true);

                // Close the new file to complete PDF creation
                toolkit.CloseOutputFile();
            }

            // Process Complete
            WriteResult("Success!");
        }

        public static void WriteResult(string result, APToolkitNET.Toolkit toolkit = null)
        {
            StringBuilder resultText = new StringBuilder();
            resultText.AppendLine(result);
            if (toolkit != null)
            {
                resultText.AppendLine($"ErrorCode: {toolkit.ExtendedErrorCode.ToString()}");
                resultText.AppendLine($"Location: {toolkit.ExtendedErrorLocation}");
                resultText.AppendLine($"Description: {toolkit.ExtendedErrorDescription}");
            }
            Console.WriteLine(resultText.ToString());
            Console.WriteLine("Press any key to exit.");
            Console.ReadKey();
        }
    }
}

File: C++/Toolkit.CreatePDF.cpp

#include <filesystem>
#include <iostream>

#import <C:/Program Files/ActivePDF/Toolkit/bin/x64/APTK.dll>

void WriteResult(const char*, int, APTOOLKITLib::IAPToolkitPtr);

int main(void)
{
    CoInitialize(NULL);

    std::filesystem::path strPath = std::filesystem::current_path();

    {
        // Toolkit COM interface pointer from aptk.tlh
        APTOOLKITLib::IAPToolkitPtr toolkitPtr;

        // Create the Toolkit instance from the ProgID
        // HKEY_CLASSES_ROOT\APToolkit.Object
        HRESULT hr = toolkitPtr.CreateInstance("APToolkit.Object");
        if (SUCCEEDED(hr))
        {
            long result = toolkitPtr->OpenOutputFile((strPath / "Toolkit.NewPDF.pdf").c_str(), 0);
            if (result == 0)
            {                
                // Each time a new page is required call NewPage
                toolkitPtr->NewPage(612.0f, 792.0f, 0);

                // Get the current version of Toolkit and save it to print on
                // the PDF
                _bstr_t toolkitVersion = toolkitPtr->ToolkitVersion;

                // Text can be added onto the new page with
                // SetFont, PrintText and PrintMultilineText functions
                toolkitPtr->SetFont("Helvetica", 24.0f, 0);
                toolkitPtr->PrintText(72.0f, 720.0f, toolkitVersion, 0);

                // Images can be added onto the new page with PrintImage,
                // PrintJPEG and PrintTIFF
                toolkitPtr->PrintJPEG(
                    (strPath / "Toolkit.Input.jpg").c_str(),
                    72.0f,
                    300.0f,
                    468.0f,
                    400.0f,
                    true,
                    0,
                    "");

                // Close the new file to complete PDF creation
                toolkitPtr->CloseOutputFile();
            }
            else
            {
                WriteResult("OpenOutputFile", result, toolkitPtr);
            }
        }
        else
        {
            std::cout << "Failed to create the Toolkit instance." << std::endl;
            std::cout << "Error: " << std::system_category().message(hr) << std::endl;
        }
    }

    CoUninitialize();

    std::cout << "Finished!" << std::endl;

    system("pause");
}

void WriteResult(const char* function, int result, APTOOLKITLib::IAPToolkitPtr toolkit)
{
    std::cout << function << " failed with error code: ";
    std::cout << result << std::endl;
    if (toolkit)
    {
        std::cout << "Extended Error Code: " << toolkit->ExtendedErrorCode << std::endl;
        std::cout << "Extended Error Description: " << toolkit->ExtendedErrorDescription << std::endl;
        std::cout << "Extended Error Location: " << toolkit->ExtendedErrorLocation << std::endl;
    }
}
